### ISSUE TITLE
Navigation properties missing

### ISSUE THREAD

**Issue by 1oglop1** [Azure/azure-sdk-for-python#1926]
Hello,
I had this problem [Stack Overflow](https://stackoverflow.com/questions/48646155/list-groups-which-has-access-to-enterprise-application-azuread-rbac)
Short story:  I was trying to list Users/Groups which are assigned to role in Application
and thanks to @lmazuel I was able to resolve the issue at least partially.

It turned out that Navigation properties are missing from sdk `azure-graphrbac`.
According to the (terrible) documentation [Azure AD Graph API Resource Path](https://msdn.microsoft.com/en-gb/library/azure/ad/graph/howto/azure-ad-graph-api-operations-overview#ResourcePath)

every resource has Navigation properties `/{resource_collection}/{resource_id}/{property_name}`

In my case it was
`{tenant_id}/servicePrincipals/{service_principal_id}/appRoleAssignedTo/`
where `appRoleAssignedTo` is the navigation property name.

**Comment by 1oglop1** [comment]
I was able to resolve this partially myself. I did create `models` for appRoleAssigned and added new operations to service principal: `role_assignments` , maybe all Navigation properties could be converted to operations.

**Comment by lmazuel** [comment]
@1oglop1 Yes, navigation properties are definitely operations (look for "owners" for instance).
Please list the navigations properties you'd like, and we'll work with @amarzavery to update the SDK.

Thanks!

**Comment by 1oglop1** [comment]
@lmazuel I noticed that code has been somehow generated therefore I did not try PR.

I would like to have navigation properties for these entities:
* Application Entity
* DirectoryObject Entity
* DirectoryRole Entity
* Group Entity
* DirectoryRole Entity
* User Entity

It would be nice to have all listed navigation properties, but in case it's not possible then most useful are operations regarding member/owner ship and application roles

**Comment by lmazuel** [comment]
@1oglop1 ownership has been added in the release I made earlier this week:
- [add_owner](https://docs.microsoft.com/en-us/python/api/azure.graphrbac.operations.applicationsoperations?view=azure-python#azure_graphrbac_operations_ApplicationsOperations_add_owner)
- [list_owners](https://docs.microsoft.com/en-us/python/api/azure.graphrbac.operations.applicationsoperations?view=azure-python#azure_graphrbac_operations_ApplicationsOperations_list_owners)

Members is already available: [get_member_groups](https://docs.microsoft.com/en-us/python/api/azure.graphrbac.operations.usersoperations?view=azure-python#azure_graphrbac_operations_UsersOperations_get_member_groups)

I don't remember for Application roles, I'll take a look.
Thanks,

**Comment by lmazuel** [comment]
And yes, this code is generated from a Swagger file:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/graphrbac/data-plane/stable/1.6/graphrbac.json

**Comment by 1oglop1** [comment]
@lmazuel Perfect thanks, it seems I'm going to learn a lot of new stuff. 
Btw are there any plans to support Microsoft Graph? There seems to be slightly more properties but it's still in Beta and I would like to use python instead of powershell :)

**Comment by lmazuel** [comment]
There is no plan on-board to support MS-Graph in near future mainly for the reason that the Azure Resource manager itself, which all SDK/CLI work with, has not integrated with MS-Graph yet.

You can however find many top awesome samples on using Python with the RestAPI directly made by @dmahugh from MSGraph
https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/concepts/python.md

**Comment by lmazuel** [comment]
Closing, since I think the discussion is over, please re-comment if I missed something.
Thank you.

**Comment by Luminger** [comment]
Is there a specific reason why the links never got implemented? I had the exact same problem as @1oglop1  earlier today and found this issue. Judging by the date I thought I missed something in the API docs, but it actually got never implemented, why's that?

I had to resort to a hacked direct call to `/users/{oid}/appRoleAssignments` to get me the data I was looking for - could we get this implemented in the SDK alongside with the other related calls? I don't see a specific reason why this was omitted tbh.

Thanks!
